server/src/main/java/org/spine3/server/storage/memory/MultitenantStorage.java

```diff
@@ -67,6 +68,20 @@ S getStorage() {
         return storage;
     }
 
+    private TenantId currentTenant() {
+        if (!isMultitenant()) {
+            return singleTenant;
+        }
+
+        final Optional<TenantId> currentTenant = CurrentTenant.get();
+
+        if (!currentTenant.isPresent()) {
+            throw new IllegalStateException("No current tenant found in multitenant execution context.");
+        }
+
+        return currentTenant.get();
+    }
+
     abstract S createSlice();
```

Comment:
As I am not able to comment in the line:45, I'll comment here.

We should probably think of concurrency here as we did in this PR for the endpoint registry.
```
  private final Map<TenantId, S> tenantSlices = newHashMap();     //  newConcurrentMap()?
```